### PR TITLE
nodePackages.clean-css: rename to clean-css-cli

### LINKED ISSUE #82527: `nodePackages.clean-css` is the wrong package
**Describe the bug**
clean-css is the NodeJS library for clean-css. clean-css-cli is the package for the clean-css cli.

**To Reproduce**
1. `nix run nixpkgs.nodePackages.clean-css`
2. cleancss ...

**Expected behavior**
The `cleancss` command should be available.

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.5.8, NixOS, 20.03beta529.da92e0566d1 (Markhor)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.2`
 - channels(jlove): `""`
 - channels(root): `"nixos-20.03beta529.da92e0566d1"`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: nodePackages.clean-css
# a list of nixos modules affected by the problem
module:
```

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

Fix #82527 . (cc @JustinLovinger ).

###### Things done

- [x] Followed https://nixos.org/nixpkgs/manual/#node.js - used `git add -p` to add only relevant updates.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by doronbehar** [comment]
I got 2 PRs' commits mixed, the current set of commits should be relevant only for clean-css-cli now.

**Review by justinlovinger — APPROVED** [review]
All files look good to me :)

**Comment by prusnak** [comment]
Please rework your PR. It now has a merge conflict after PR https://github.com/NixOS/nixpkgs/pull/89184 has been merged

**Comment by doronbehar** [comment]
Fixed merged conflicts.

File: pkgs/development/node-packages/node-packages.nix [hunk 1]
```diff
@@ -53570,21 +53570,34 @@ in
...
+      sources."path-is-absolute-1.0.1"
       sources."source-map-0.6.1"
+      sources."wrappy-1.0.2"
     ];
     buildInputs = globalBuildInputs;
     meta = {
-      description = "A well-tested CSS minifier";
-      homepage = https://github.com/jakubpawlowicz/clean-css;
+      description = "A command-line interface to clean-css CSS optimization library";
+      homepage = "https://github.com/jakubpawlowicz/clean-css-cli#readme";
```
Comment (prusnak) [inline pkgs/development/node-packages/node-packages.nix:53600]: nitpick: `#readme` at the end is redundant

**Review by prusnak — APPROVED** [review]

File: pkgs/development/node-packages/node-packages.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/development/node-packages/node-packages.nix:53600]: @prusnak I didn't write this PR by hand - I ran `./generate.sh` after changing `clean-css` to `clean-css-cli` and I cherry-picked this change. This `#readme` is how upstream declares it in their `package.json`, just as all other meta info was taken from there.

Comment (doronbehar) [reply pkgs/development/node-packages/node-packages.nix:53600]: BTW FYI, running `./generate.sh` took me about half a day. It took me this long to do it initially - before the merge conflicts.

Comment (prusnak) [reply pkgs/development/node-packages/node-packages.nix:53600]: OK then.

**Comment by doronbehar** [comment]
@GrahamcOfBorg nodePackages.clean-css-cli

**Comment by doronbehar** [comment]
@GrahamcOfBorg build nodePackages.clean-css-cli

**Comment by veprbl** [comment]
We likely may need to do something with base64 hashes (e.g. apply #89520). Meanwhile, I think, the PR could stick to changing node-packages.json file only.

**Comment by doronbehar** [comment]
@veprbl how often does someone run `./generate.sh` for the node package set?

**Comment by veprbl** [comment]
Landed as 9a47504fe7cff237d8ecb1c87b21b601ef03f335

> @veprbl how often does someone run `./generate.sh` for the node package set?

If you are asking for a suggestions, I'm not sure what to tell you. While I was applying this PR and another one, some other upstream dependencies had already been bumped. The way it is now, any PR that regenerates node-packages.nix will pick up irrelevant changes that would always lead to merge conflicts. Perhaps, it would work a bit more predictably if node2nix could fetch some specific snapshot of the npm registry.

**Pull request closed without merging**